packages/material-ui/src/Tabs/ScrollbarSize.js

```diff
@@ -18,53 +17,44 @@ const styles = {
  * The component is originates from https://github.com/STORIS/react-scrollbar-size.
  * It has been moved into the core in order to minimize the bundle size.
  */
-class ScrollbarSize extends React.Component {
-  constructor() {
-    super();
+function ScrollbarSize(props) {
+  const { onChange } = props;
+  const scrollbarHeight = React.useRef();
+  const nodeRef = React.useRef();
 
-    if (typeof window !== 'undefined') {
-      this.handleResize = debounce(() => {
-        const prevHeight = this.scrollbarHeight;
-        this.setMeasurements();
-
-        if (prevHeight !== this.scrollbarHeight) {
-          this.props.onChange(this.scrollbarHeight);
-        }
-      }, 166); // Corresponds to 10 frames at 60 Hz.
+  const setMeasurements = () => {
+    if (!nodeRef.current) {
```

Comment:
Do we need this logic?

Reply:
I guess this is just some defensive logic if this somehow gets executed after unmount. I believe we have extra defensive logic against execution after unmount in other places too.

Reply:
Can we try to remove it? See if it fails the CI or if somebody reports an issue? The history of this branch lead me to: https://github.com/mui-org/material-ui/pull/11523.

Reply:
It's also not part of the original component:
https://github.com/STORIS/react-scrollbar-size/blob/fd8ae94f9ad39b8c45109e88971b996a72353e29/src/ScrollbarSize.jsx#L39-L42

Don't know why you want to remove it.

Reply:
Oh perfect, so it's something I added that I now want to remove. It looks like I changed my mind. Why do I want to remove it? I don't believe in defensive checks. Most of the time, they hide something. I would like to know if it hides something, or if we can just remove it. I can't think of any reason a defensive check is useful here. Do you? Humm, maybe I added it to reduce the pain of shallow testing, not a good reason!

Reply:
> Humm, maybe I added it to reduce the pain of shallow testing, not a good reason!

Sounds about right. Maybe `debounce` flushes queued callbacks asynchronously when calling `cancel`?

Reply:
We have a clear() method for this problem.